pkg/controller/direct/references/projectref.go

```diff
@@ -88,3 +89,18 @@ func ResolveProject(ctx context.Context, reader client.Reader, src client.Object
 		ProjectID: projectID,
 	}, nil
 }
+
+func ResolveProjectForTFGenerated(ctx context.Context, reader client.Reader, src client.Object, ref *v1alpha1.ResourceRef) (*Project, error) {
+	if ref == nil {
+		return nil, nil
+	}
+	if ref.Kind != "Project" {
+		return nil, fmt.Errorf("resolve project: %s|%s|%s", ref.Kind, ref.Name, ref.Name)
```

Comment:
ref.Name is repeated.  And maybe we should have a clearer message, like "expected kind to be Project, got %q"